### PR TITLE
Make auto-flushes opt-in only

### PR DESCRIPTION
**Pull request by casmith**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,6 +50,21 @@ Lastly be sure to load the custom Jasmine matchers before your tests run like so
...
 
+If you would like the promise matchers to flush $httpBackend, $interval, or $timeout automatically, some options are available:
+
+```javascript
+  beforeEach(function() {
+
+    installPromiseMatchers({
+      httpBackend: true,
+      interval: true,
+      timeout: true
```
Comment (bvaughn) [inline README.md]: nit: I think `flushHttpBackend` would be a more descriptive name than `httpBackend`

Comment (bvaughn) [reply README.md]: I also think we should leave these on by default for now, so we don't have to release a new major.

File: tests/test.js [hunk 2]
```diff
@@ -13,7 +13,11 @@ describe('Promise Matcher tests', function () {
   beforeEach(function() {
     angular.mock.module('foobar');
 
-    installPromiseMatchers();
+    installPromiseMatchers({
+      httpBackend: true,
+      interval: true,
+      timeout: true
+    });
```
Comment (bvaughn) [inline tests/test.js]: Would be nice to have a test that verifies the newly added behavior (e.g. that flush can be disabled)

**Comment by bvaughn** [comment]
Hi @casmith 👋

Thanks for another PR. You've contributed a few PRs to this project over the years, so I've decided to add you as a collaborator.

File: README.md (hunk 1 above)
Comment (casmith) [reply README.md:61]: Good point - should have run this by the discussion before opening the PR.

Comment (bvaughn) [reply README.md:61]: No worries 😄 I changed it before merging

**Pull request merged**